### DISCUSSION TITLE
Maintaining subscription when client goes down

### DISCUSSION THREAD

**Discussion by SangavMut** [FreeOpcUa/opcua-asyncio#1667] [Q&A]
Right now in asyncua , when i create a subscription and the client goes down, the subscription gets deleted and the data is lost during client down time. Is there any way to not lose data when the client is down? what are all the possible solutions.

![image (1)](https://github.com/FreeOpcUa/opcua-asyncio/assets/109574334/71ae51b6-5bd3-4602-a0ae-44d020ac7e02)

**Comment by schroeder-** [comment]
The feature want is [Durable Subscriptions](https://reference.opcfoundation.org/Core/Part4/v104/docs/6.8). 

It depends if you use asyncua as client, you depend that the server supports durable subscriptions, the client side must only call some methods and set parameter the correct way.

Asyncua as server is more complicate, but could be doable. Still your clients has to support setting all parameters and call the methods, need to make a subscription durable.

**Reply by oroulet** [reply]
What we do have is server history. If you enable history you can fetch events you missed

**Reply by SangavMut** [reply]
is there any other way than durable subscription and server history, because both of them have to be supported by the server?
